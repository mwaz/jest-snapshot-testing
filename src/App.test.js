import React from "react";
import renderer from "react-test-renderer";
import App from "./App";
import Counter from "./Counter";

describe("Jest Snapshot testing suite", () => {
  it("matches snapshot", () => {
    const domTree = renderer.create(<App />).toJSON();
    expect(domTree).toMatchSnapshot();
  });

  it("Should render 3 after three increments", () => {
    const component = renderer.create(<Counter />);
    component.getInstance().increment();
    component.getInstance().increment();
    component.getInstance().increment();
    expect(component.toJSON()).toMatchSnapshot();
  });
});
