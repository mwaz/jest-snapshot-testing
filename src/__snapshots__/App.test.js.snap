// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Jest Snapshot testing suite Should render 3 after three increments 1`] = `
<div
  className="counter"
>
  <h1>
    COUNTER
  </h1>
  <div
    className="buttons"
  >
    <button
      onClick={[Function]}
    >
      Increment
    </button>
    <button
      onClick={[Function]}
    >
      Decrement
    </button>
  </div>
  <p>
    3
  </p>
</div>
`;

exports[`Jest Snapshot testing suite Should render count decrements 1`] = `
<div
  className="counter"
>
  <h1>
    COUNTER
  </h1>
  <div
    className="buttons"
  >
    <button
      onClick={[Function]}
    >
      Increment
    </button>
    <button
      onClick={[Function]}
    >
      Decrement
    </button>
  </div>
  <p>
    1
  </p>
</div>
`;

exports[`Jest Snapshot testing suite matches snapshot 1`] = `
<div
  className="App"
>
  <div
    className="counter"
  >
    <h1>
      COUNTER
    </h1>
    <div
      className="buttons"
    >
      <button
        onClick={[Function]}
      >
        Increment
      </button>
      <button
        onClick={[Function]}
      >
        Decrement
      </button>
    </div>
    <p>
      0
    </p>
  </div>
</div>
`;
